src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/AuthModal.java

```diff
@@ -42,14 +48,31 @@ private void switchToMainWindowHandle() {
 
   public boolean isOpened() {
     switchToAuthModalHandle();
-    boolean isOpenedResult = authModal.isDisplayed();
+    try {
+      wait.forElementVisible(registerAuthModal);
+    } catch (NoSuchElementException e) {
+      PageObjectLogging.logInfo("Register Auth Modal is not displayed");
+      return false;
+    }
     switchToMainWindowHandle();
-    return isOpenedResult;
+    return true;
+  }
+
+  public boolean isSignInOpened() {
+    switchToAuthModalHandle();
+    try {
+      wait.forElementVisible(signInAuthModal);
+    } catch (NoSuchElementException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)